analytics/analytics/job_processor/artifacts.py

```diff
@@ -37,7 +43,7 @@ def get_job_artifacts_file(job: ProjectJob, filename: str):
             with open(artifacts_file, "wb") as f:
                 job.artifacts(streamed=True, action=f.write)
         except GitlabGetError:
-            raise JobArtifactFileNotFound(job, filename)
+            raise JobHasNoArtifacts(job)
```

Comment:
How can we confidently raise this error here, as opposed to the first one? We don't really know that the job has no artifacts based on this exception, do we?

Reply:
Good point. I renamed it to `JobArtifactDownloadFailed`

Reply:
Also, the existing `JobArtifactFileNotFound` exception is caught in https://github.com/spack/spack-infrastructure/blob/d5df819cdf1a7beef5e0e0e7cd0fd37276404681/analytics/analytics/job_processor/metadata.py#L183

which will re-raise the error if the job status isn't `failed`. So we'd also need to update that.

Reply:
Updated